app/scripts/services/authorization.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, Logger, $interval, DataService, APIService){
+    function AuthorizationService() {
+      this.rules = null;
+      this.intervalTimer = null;
+      this.forceReload = false;
+      this.canICreateSomeResource = false;
+      this.projectNameRules = null;
+    }
+
+    var normalizeRules = function(rules) {
+      var normalizedRules = {};
+      _.each(rules, function(rule) {
+        _.each(rule.resources, function(resource) {
+          normalizedRules[resource] = rule.verbs;
+        });
+      });
+      return normalizedRules;
+    };
+
+    AuthorizationService.prototype.getProjectRules = function(projectName) {
+      var self = this;
+      if (!self.rules || projectName !== self.projectNameRules) {
```

Comment:
@liggitt dont think I follow your concern. The implementation caches the rules for current project, where the ProjectsService will determin which project rules are going to be loaded. When changing the project the rules for the other project will be loaded and cached..
For checking permissions in other projects we should use `subjectaccessreviews` call on specific resource-verb-namespace(dont have the method for that here since I was using it for checking delete permissions of the projects on the projects listing page, but we agreed with @jwforres that we wont do that until we have paging since the page could list lots of projects, which could lead to lots of api calls)